sage.py

```diff
@@ -571,393 +485,271 @@ def getepisodes(action_seq, mcat, plot, debug=False):
-                plt.title('Micro attack episodes | Team: '+str(tid) +' | Host: '+ '->'.join([x for x in attacker]))
+                plt.title('Micro attack episodes | Team: ' + str(tid) + ' | Host: ' + '->'.join(attacker_victim))
                 plt.xlabel('Time Window (sec)')
                 plt.ylabel('Micro attack stages')
-                # NOTE: Line thicknesses are on per host basis
-                tmax= max([x[4] for x in h_ep])
-                tmin = min([x[4] for x in h_ep])
-                for idx, ep in enumerate(h_ep):
-                    #print(idx, (ep[0], ep[1]), ep[2], ep[3][ep[0]:ep[1]+1])
-                    xax = list(range(ep[0], ep[1]+1))
-                    yax = [mcats.index(ep[2])]*len(xax)
-                    thickness  = ep[4]
-                    lsize = ((thickness - tmin)/ (tmax - tmin)) * (5-0.5) + 0.5 if (tmax - tmin) != 0.0 else 0.5
-                    #lsize = np.log(thickness) + 1 TODO: Either take log or normalize between [0.5 5]
-                    msize = (lsize*2)+1
+                # NOTE: Line thicknesses are on per-host basis
+                tmax = max([epi[4] for epi in host_episodes])
+                tmin = min([epi[4] for epi in host_episodes])
+                for idx, ep in enumerate(host_episodes):
+                    xax = list(np.arange(ep[0], ep[1]+1))
+                    yax = [mcats.index(ep[2])] * len(xax)
+                    thickness = ep[4]
+                    lsize = ((thickness - tmin) / (tmax - tmin)) * (5 - 0.5) + 0.5 if (tmax - tmin) != 0.0 else 0.5
+                    # lsize = np.log(thickness) + 1 TODO: Either take log or normalize between [0.5 5]
+                    msize = (lsize * 2) + 1
                     ax.plot(xax, yax, color=mcols[macro_inv[micro2macro[micro[ep[2]]]]], linewidth=lsize)
                     ax.plot(ep[0], mcats.index(ep[2]), color=mcols[macro_inv[micro2macro[micro[ep[2]]]]], marker='.', linewidth=0, markersize=msize, label=micro2macro[micro[ep[2]]])
                     ax.plot(ep[1], mcats.index(ep[2]), color=mcols[macro_inv[micro2macro[micro[ep[2]]]]], marker='.', linewidth=0, markersize=msize)
-                    plt.yticks(range(len(mcats)), mcats_lab, rotation='0')
+                    plt.yticks(range(len(mcats)), [x.split('.')[1] for x in micro.values()], rotation=0)
                 legend_without_duplicate_labels(ax)
                 plt.grid(True, alpha=0.4)
                 
-                #plt.tight_layout()
-                #plt.savefig('Pres-Micro-attack-episodes-Team'+str(tid) +'-Connection'+ attacker[0]+'--'+attacker[1]+'.png')
+                # plt.tight_layout()
+                # plt.savefig('Pres-Micro-attack-episodes-Team'+str(tid) +'-Connection'+ attacker[0]+'--'+attacker[1]+'.png')
                 plt.show()
         
-        team_episodes.append(t_ep)
+        team_episodes_.append(team_host_episodes)
         team_times.append(_team_times)
-    return (team_episodes, team_times)
```

Comment:
`team_times` are not used anywhere else in the code. They are ignored when the method returns. Should I still keep them or can they be deleted?

Reply:
Keep them. They store the first activation time for each team. Might be useful at some point.